server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotShutdownIT.java

```diff
@@ -540,12 +540,13 @@ public void testSnapshotShutdownProgressTracker() throws Exception {
         mockLog.awaitAllExpectationsMatched();
         resetMockLog();
 
+        assert 1 <= numShards && numShards <= 10;
         mockLog.addExpectation(
-            new MockLog.SeenEventExpectation(
+            new MockLog.PatternSeenEventExpectation(
                 "SnapshotShutdownProgressTracker running number of snapshots",
                 SnapshotShutdownProgressTracker.class.getCanonicalName(),
                 Level.INFO,
-                "*Number shard snapshots running [" + numShards + "].*"
+                ".+Number shard snapshots running \\[" + (numShards < 10 ? "[1-" + numShards + "]" : "([1-9]|10)") + "].+"
```

Comment:
Can we change the `final int numShards = randomIntBetween(1, 10);` to `final int numShards = randomIntBetween(1, 9);` to simplify this? The random number range is arbitrary, so I think that would be okay.

Reply:
++ (or else maybe leave this as a `SeenEventExpectation` and parse out the `numShards` by overriding `innerMatch` rather than trying to do anything overly clever with a regex)